File: iosApp/iosApp/screens/onboarding/LoginScreen.swift
import SwiftUI
import Routing
import KMPObservableViewModelSwiftUI
import shared

struct LoginScreen: View {
    @StateViewModel var viewModel = ToDoViewModel(toDoRepository: KoinDependencies().toDoRepository)
    @State private var email = "emilys"
    @State private var password = "emilyspass"
    @StateObject var router: Router<MainNavigation>
    @State private var loginError = false
    
    init(router: Router<MainNavigation>) {
        _router = StateObject(wrappedValue: router)
    }
    
    var body: some View {
        BaseScreen(content: {
            VStack(spacing: 0) {
                HStack{
                    Image(resource: \.vec_ic_back) // Replace with actual icon name if using asset
                        .resizable()
                        .frame(width: 44, height: 44)
                        .padding(.top, 60)
                   
                }.frame(maxWidth: .infinity, alignment: .leading)
                // Back button (Image)
                
                
                Spacer().frame(height: 60)
                
                // Header
                Text(Strings().get(id: SharedRes.strings().login_label_title, args: []))
                    .font(.title)
                    .fontWeight(.bold)
                    .padding(.horizontal)
                
                Spacer().frame(height: 40)
                
                // Email input
                InputText(label: Strings().get(id: SharedRes.strings().login_label_email, args: []), text: $email, onTextChanged: { _ in }, icon: \.vec_ic_email, secure: false).frame(maxWidth: .infinity)
                
                Spacer().frame(height: 14)
                
                // Password input
                InputText(label: Strings().get(id: SharedRes.strings().login_label_password, args: []), text: $password, onTextChanged: { _ in}, icon: \.vec_ic_lock, secure: true).frame(maxWidth: .infinity)
                
                Spacer().frame(height: 24)
                
                // Login button (Primary)
                ButtonPrimary(text: Strings().get(id: SharedRes.strings().button_login, args: []), action: {
                    DispatchQueue.main.asyncAfter(deadline: .now()) {
                        viewModel.login(username: email, password: password, completionHandler: {response,_ in
                            if let result = response {
                                let data = result as LoginResult
                                if !data.username.isEmpty  {
                                    loginError = false
                                    router.routeTo(.dashboard)
                                } else {
                                    loginError = true
                                }
                            }
                        })
                    }
                }).frame(maxWidth: .infinity)
                
                Spacer().frame(height: 24)
                
                // Divider with "or" in between
                HStack {
                    Divider().background(Color.gray).frame(height: 1)
                    Text(Strings().get(id: SharedRes.strings().separator_or, args: []))
                        .font(.body)
                        .padding(.horizontal, 4)
                    Divider().background(Color.gray).frame(height: 1)
                }
                .padding(.horizontal)
                
                Spacer().frame(height: 24)
                
                // Social Login buttons (Google and Apple)
                ButtonSocial(text: Strings().get(id: SharedRes.strings().button_social_google, args: []), action: {}, icon: \.vec_ic_google)
                Spacer().frame(height: 12)
                ButtonSocial(text: Strings().get(id: SharedRes.strings().button_social_apple, args: []), action: {}, icon: \.vec_ic_apple)
            }
            .padding(4)
            .background(Color.white) // Background color for the screen
            .edgesIgnoringSafeArea(.all)
        })
    }
}


struct LoginScreen_Preview: PreviewProvider {
    static var previews: some View {
        RoutingView(MainNavigation.self) { router in
            LoginScreen(router: router)
                .previewLayout(.sizeThatFits)
                .padding()
        }
    }
}

File: androidApp/src/main/java/com/todo/app/android/screens/onboarding/LoginScreen.kt
package com.todo.app.android.screens.onboarding

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.todo.app.SharedRes
import com.todo.app.android.screens.BaseScreen
import com.todo.app.android.screens.component.ButtonPrimary
import com.todo.app.android.screens.component.ButtonSocial
import com.todo.app.android.screens.component.InputText
import com.todo.app.android.screens.component.TextBody
import com.todo.app.android.screens.component.TextTitle
import com.todo.app.android.stringResource
import com.todo.app.viewmodel.ToDoViewModel
import kotlinx.coroutines.launch
import org.koin.androidx.compose.koinViewModel

@Composable
fun LoginScreen(toDashboard: () -> Unit) {
    var username by remember { mutableStateOf("emilys") }
    var password by remember { mutableStateOf("emilyspass") }
    val viewModel = koinViewModel<ToDoViewModel>()
    val scope = rememberCoroutineScope()
    var loginError by remember { mutableStateOf(false) }
    BaseScreen(composable = {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(Color.White),
        ) {
            Image(
                modifier = Modifier
                    .width(44.dp)
                    .height(44.dp),
                painter = painterResource(com.todo.app.R.drawable.vec_ic_back),
                contentDescription = null
            )
            Spacer(Modifier.height(60.dp))
            TextTitle(text = stringResource(id = SharedRes.strings.login_label_title))
            Spacer(Modifier.height(40.dp))
            InputText(
                icon = com.todo.app.R.drawable.vec_ic_email,
                label = stringResource(id = SharedRes.strings.login_label_email),
                text = username,
                onTextChanged = { username = it })
            Spacer(Modifier.height(14.dp))
            InputText(
                secure = true,
                icon = com.todo.app.R.drawable.vec_ic_lock,
                label = stringResource(id = SharedRes.strings.login_label_password),
                text = password,
                onTextChanged = { password = it })

            Spacer(Modifier.height(24.dp))
            ButtonPrimary(text = "Log in", action = {
                scope.launch {
                    val result = viewModel.login(username, password)
                    if (result.username.isNotEmpty()) {
                        loginError = false
                        toDashboard()
                    } else {
                        loginError = true
                    }
                }
            })
            Spacer(Modifier.height(24.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                        .height(1.dp)
                        .background(Color.Gray)
                )
                Spacer(Modifier.width(4.dp))
                TextBody("or")
                Spacer(Modifier.width(4.dp))
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                        .height(1.dp)
                        .background(Color.Gray)
                )
            }
            Spacer(Modifier.height(24.dp))
            ButtonSocial(
                text = stringResource(id = SharedRes.strings.button_social_google),
                icon = com.todo.app.R.drawable.vec_ic_google,
                action = {})
            Spacer(Modifier.height(12.dp))
            ButtonSocial(
                text = stringResource(id = SharedRes.strings.button_social_apple),
                icon = com.todo.app.R.drawable.vec_ic_apple,
                action = {})
        }
    })
}

@Preview
@Composable
fun LoginScreen_Preview() {
    LoginScreen(toDashboard = {})
}